crates/utils/src/lib.rs

```diff
@@ -211,6 +209,8 @@ pub mod tokio_lock;
 pub mod tracing;
 
 use ::tracing::debug;
+/// Interval tree implementation
+pub use interval_map;
```

Comment:
We could directly use interval_map in xline crate.

Reply:
That would require changing the dependencies of two crates, `xline` and `xlineapi`. Is that okay? Now all we need to do is modify the `utils`.